pkg/healthcheck/healthcheck.go

```diff
@@ -10,70 +10,34 @@ import (
 	"time"
 
 	"github.com/sirupsen/logrus"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/client-go/tools/clientcmd/api"
-	"k8s.io/client-go/tools/clientcmd/api/latest"
-	"k8s.io/client-go/tools/clientcmd/api/v1"
 
-	acsapi "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/api"
 	"github.com/openshift/openshift-azure/pkg/log"
 )
 
-// GetKubeconfigFromV1Config takes a v1 config and returns a kubeconfig
-func getKubeconfigFromV1Config(kc *v1.Config) (clientcmd.ClientConfig, error) {
-	var c api.Config
-	err := latest.Scheme.Convert(kc, &c, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	kubeconfig := clientcmd.NewDefaultClientConfig(c, &clientcmd.ConfigOverrides{})
-
-	return kubeconfig, nil
-}
-
 type HealthChecker interface {
-	HealthCheck(ctx context.Context, cs *acsapi.OpenShiftManagedCluster) error
+	HealthCheck(ctx context.Context, cs *api.OpenShiftManagedCluster) error
 }
 
 type simpleHealthChecker struct {
-	pluginConfig acsapi.PluginConfig
+	pluginConfig api.PluginConfig
 }
 
 var _ HealthChecker = &simpleHealthChecker{}
 
 // NewSimpleHealthChecker create a new HealthChecker
-func NewSimpleHealthChecker(entry *logrus.Entry, pluginConfig acsapi.PluginConfig) HealthChecker {
+func NewSimpleHealthChecker(entry *logrus.Entry, pluginConfig api.PluginConfig) HealthChecker {
 	log.New(entry)
 	return &simpleHealthChecker{pluginConfig: pluginConfig}
 }
 
 // HealthCheck function to verify cluster health
-func (hc *simpleHealthChecker) HealthCheck(ctx context.Context, cs *acsapi.OpenShiftManagedCluster) error {
-	kc, err := newClientSet(ctx, cs.Config.AdminKubeconfig)
-	if err != nil {
-		return err
-	}
-
-	// ensure that all nodes are ready
-	err = WaitForNodes(ctx, cs, kc)
-	if err != nil {
-		return err
-	}
-
-	// Wait for infrastructure services to be healthy
-	err = WaitForInfraServices(ctx, kc)
-	if err != nil {
-		return err
-	}
-
+func (hc *simpleHealthChecker) HealthCheck(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
```

Comment:
think we can entirely get rid of this package and put the console check at the end of plugin.CreateOrUpdate()

Reply:
done